registry/noop_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"testing"
+
+	"github.com/kubernetes-incubator/external-dns/endpoint"
+	"github.com/kubernetes-incubator/external-dns/internal/testutils"
+	"github.com/kubernetes-incubator/external-dns/plan"
+	"github.com/kubernetes-incubator/external-dns/provider"
+)
+
+var _ Registry = &NoopRegistry{}
+
+func TestNoopRegistry(t *testing.T) {
+	t.Run("NewNoopRegistry", testNoopInit)
+	t.Run("Records", testNoopRecords)
+	t.Run("ApplyChanges", testNoopApplyChanges)
+}
+
+func testNoopInit(t *testing.T) {
+	p := provider.NewInMemoryProvider()
+	r, err := NewNoopRegistry(p)
+	if err != nil {
+		t.Error(err)
```

Comment:
This should probably be `t.Fatal` because you'll get a null pointer error on the next `if`